File: app/src/main/java/com/example/foodshare/HomeActivity.java
package com.example.foodshare;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.foodshare.models.Restaurant;
import com.example.tictactoe.R;

import java.util.ArrayList;
import java.util.List;

public class HomeActivity extends AppCompatActivity {

    private RestaurantAdapter restaurantAdapter;
    private List<Restaurant> restaurantList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);

        // Initialize the RecyclerView and its adapter
        RecyclerView restaurantRecyclerView = findViewById(R.id.restaurantRecyclerView);
        restaurantList = new ArrayList<>();
        restaurantAdapter = new RestaurantAdapter(restaurantList);

        // Set the layout manager and adapter for the RecyclerView
        restaurantRecyclerView.setLayoutManager(new LinearLayoutManager(this));
        restaurantRecyclerView.setAdapter(restaurantAdapter);

        // Populate the restaurant list (Replace with your actual data retrieval logic)
        populateRestaurantList();

        // Set a click listener for the quantity available link
        restaurantAdapter.setOnQuantityAvailableClickListener(new RestaurantAdapter.OnQuantityAvailableClickListener() {
            @Override
            public void onQuantityAvailableClick(int position) {
                // Handle the click event, e.g., navigate to the restaurant details page
                Restaurant restaurant = restaurantList.get(position);
                Intent intent = new Intent(HomeActivity.this, RestaurantDetailsActivity.class);
                intent.putExtra("restaurant", restaurant);
                startActivity(intent);
            }
        });
    }

    @SuppressLint("NotifyDataSetChanged")
    private void populateRestaurantList() {
        // Fetch the restaurant data for the donation center user from the database
        // and add it to the restaurantList

        // Example data
        restaurantList.add(new Restaurant(1, "Restaurant A", 10, "123 Main St", "restaurant_image_1", "4.5", "Good food"));
        restaurantList.add(new Restaurant(2, "Restaurant B", 5, "456 Elm St", "restaurant_image_2", "3.8", "Cozy atmosphere"));
        restaurantList.add(new Restaurant(3, "Restaurant C", 8, "789 Oak St", "restaurant_image_3", "4.2", "Friendly staff"));

        // Notify the adapter that the data set has changed
        restaurantAdapter.notifyDataSetChanged();
    }
}

File: app/src/main/java/com/example/foodshare/HomeActivity.kt
package com.example.foodshare

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.foodshare.models.Restaurant
import com.example.tictactoe.R

class HomeActivity : AppCompatActivity() {
    private lateinit var restaurantList: RecyclerView
    private lateinit var restaurantAdapter: RestaurantAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)

        restaurantList = findViewById(R.id.restaurantList)
        restaurantAdapter = RestaurantAdapter(emptyList()) { restaurant ->
            // Handle restaurant item click event
            navigateToRestaurantDetails(restaurant)
        }
        restaurantList.layoutManager = LinearLayoutManager(this)
        restaurantList.adapter = restaurantAdapter

        // Populate restaurant data (replace with your data)
        val restaurants = listOf(
            Restaurant("Restaurant 1", "10", "Address 1", "4.5", "Good food"),
            Restaurant("Restaurant 2", "5", "Address 2", "4.2", "Nice ambiance")
        )
        restaurantAdapter.updateRestaurants(restaurants)
    }

    private fun navigateToRestaurantDetails(restaurant: Restaurant) {
        val intent = Intent(this, RestaurantDetailsActivity::class.java)
        intent.putExtra("restaurant", restaurant)
        startActivity(intent)
    }
}